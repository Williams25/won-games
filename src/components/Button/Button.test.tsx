import "jest-styled-components";
import { screen } from "@testing-library/react";
import { Button } from "./index";
import { renderWithTheme } from "utils/tests/helpers";
import { AddShoppingCart } from "@styled-icons/material-rounded/AddShoppingCart";

describe("Button", () => {
  it("should render the medium size by default", () => {
    const { container } = renderWithTheme(<Button>Buy now</Button>);

    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      height: "4rem",
      padding: "0.8rem 3.2rem",
      "font-size": "1.4rem"
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the large size", () => {
    renderWithTheme(<Button size="large">Buy now</Button>);

    expect(screen.getByRole("button", { name: /Buy now/ })).toHaveStyle({
      height: "5rem",
      padding: "0.8rem 4.8rem",
      "font-size": "1.6rem"
    });
  });

  it("should render the small size", () => {
    renderWithTheme(<Button size="small">Buy now</Button>);

    expect(screen.getByRole("button", { name: /Buy now/ })).toHaveStyle({
      height: "3rem",
      padding: "0.8rem",
      "font-size": "1.2rem"
    });
  });

  it("should render the fullWidth size", () => {
    renderWithTheme(<Button fullWidth={true}>Buy now</Button>);

    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      width: "100%"
    });
  });

  it("should darken the color with the focus", () => {
    renderWithTheme(<Button>Buy now</Button>);

    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyleRule(
      "filter",
      "brightness(0.9)",
      {
        modifier: ":hover"
      }
    );
  });

  it("should render an icon version", () => {
    renderWithTheme(
      <Button fullWidth icon={<AddShoppingCart data-testid="icon" />}>
        Buy now
      </Button>
    );

    expect(screen.getByText(/Buy now/i)).toBeInTheDocument();
    expect(screen.getByTestId(/icon/i)).toBeInTheDocument();
  });
});
