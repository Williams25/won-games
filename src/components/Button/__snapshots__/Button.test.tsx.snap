// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the medium size by default 1`] = `
.c0 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  padding: 0.8rem 3.2rem;
  height: 4rem;
  font-size: 1.4rem;
}

.c0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

<button
  class="c0"
>
  <span>
    Buy now
  </span>
</button>
`;
